File: frontend/src/types/website-intelligence.ts
export interface WebsiteIntelligence {
  id: string;
  tenant_id: string;
  website_url: string;
  scrape_status: "pending" | "in_progress" | "completed" | "failed" | "skipped";
  analysis_result: AnalysisResult | null;
  summary: string | null;
  suggestions: WebsiteSuggestion[];
  applied_to_onboarding: boolean;
  input_tokens: number | null;
  output_tokens: number | null;
  estimated_cost: number | null;
  error_message: string | null;
  created_at: string;
}

export interface AnalysisResult {
  product_lines: Record<string, DetectionResult>;
  vault_lines: Record<string, DetectionResult>;
  certifications: {
    npca_certified: DetectionResult;
    wilbert_licensee: DetectionResult;
    other_certifications: string[];
  };
  spring_burials: DetectionResult;
  urn_categories: Record<string, DetectionResult>;
  summary: string;
}

export interface DetectionResult {
  detected: boolean;
  confidence: number;
  evidence: string | null;
}

export interface WebsiteSuggestion {
  id: string;
  suggestion_type: string;
  suggestion_key: string;
  suggestion_label: string;
  confidence: number;
  confidence_label: string;
  evidence: string | null;
  status: "pending" | "accepted" | "dismissed";
}

export interface SuggestedProduct {
  product_key: string;
  confidence: number;
  should_precheck: boolean;
  should_highlight: boolean;
}

File: backend/app/schemas/website_intelligence.py
"""Schemas for Website Intelligence feature."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SuggestionResponse(BaseModel):
    id: str
    suggestion_type: str
    suggestion_key: str
    suggestion_label: str
    confidence: float
    evidence: str | None
    status: str
    confidence_label: str  # computed: "High confidence" / "Likely" / "Possible"

    model_config = ConfigDict(from_attributes=True)


class WebsiteIntelligenceResponse(BaseModel):
    id: str
    tenant_id: str
    website_url: str
    scrape_status: str
    analysis_result: dict | None
    suggestions: list[SuggestionResponse]
    summary: str | None
    input_tokens: int | None
    output_tokens: int | None
    estimated_cost: float | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SuggestionUpdateRequest(BaseModel):
    status: str  # "accepted" or "dismissed"


class WebsiteIntelligenceCreate(BaseModel):
    website_url: str
